### DISCUSSION TITLE
API rules. Is there a way to view a record by knowing the id?

### DISCUSSION THREAD

**Discussion by GalaxysHub** [pocketbase/pocketbase#3932] [Q&A] [answered]
I'm trying to allow view access on a record if the user knows the id. 

I'm able to update the id with the rule `@request.data.id = id` but this doesn't work on the view field because there's no data object passed to the request. I'm using `pb.collections("collection").getOne(id)` to query the collection. I've also tried `@request.query.id = id` but it doesn't work either

**Comment by ganigeorgiev** [comment]
I don't understand. If you want everyone to be able to view a record by its id, then you can leave the view rule empty.

**Reply by GalaxysHub** [reply]
Thanks for your reply. I'm still new to pocketbase, so I still don't fully understand its functionality. But in this case, I don't want someone to be able to run a query like `pb.collection("collection").getFullList()` and query the entire collection. 

In our case, each user has his own record assigned to him that he should be able to access when he's given the id. In addition, we don't want the users to have accounts.

**Reply by ganigeorgiev** [reply, marked as answer]
List and View rules are separate.

If you want someone **only** to be able to access a record by its id, then you can "lock" the List API rule (aka. only admins will be able to list/search) and "open" (aka. leave empty) the View API rule for everyone who know the id of the record.
